android/app/src/main/java/app/organicmaps/util/Utils.java

```diff
@@ -347,13 +347,10 @@ public static void sendFeedback(@NonNull Activity activity)
                                                                  Constants.Email.SUPPORT));
   }
 
-  public static void navigateToParent(@Nullable Activity activity)
+  public static void navigateToParent(@NonNull Activity activity)
   {
-    if (activity == null)
-      return;
-
-    if (activity instanceof CustomNavigateUpListener)
-      ((CustomNavigateUpListener) activity).customOnNavigateUp();
+    if (activity instanceof MwmActivity)
```

Comment:
A comment why this custom logic is needed and when it happens would be helpful.

Reply:
It seems the initial concept was for Fragments like EditorFragment, SearchFragment, and others to run within their dedicated activities (EditorActivity, SearchActivity, etc.) in phone mode or as part of MwmActivity in tablet mode. I have doubts that dedicated activities are needed at all, but this is a different topic to research that can handled as part of table UI improvement0. 

As part of changes done by this PR, MwmActivity was the only class that implemented CustomNavigateUpListener. Therefore, I simply removed the interface without changing the navigation logic.